File: app/src/main/java/com/example/interactions/about/AboutActivity.java
package com.example.interactions.about;

import android.content.Context;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.MenuItem;
import android.view.View;

import com.example.R;
import com.example.android.AppUtils;
import com.example.databinding.AboutActivityBinding;
import com.example.interactions.AbstractActivity;

import de.psdev.licensesdialog.LicensesDialog;

public class AboutActivity extends AbstractActivity {
	AboutActivityBinding binding;

	@NonNull
	public static Intent newIntent(@NonNull Context context) {
		return new Intent(context, AboutActivity.class);
	}

	@NonNull
	@Override
	protected View root() {
		return binding.getRoot();
	}

	@Override
	protected void onCreate(@Nullable Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		binding = DataBindingUtil.setContentView(this, R.layout.about_activity);

		setUpToolBar(binding.toolbar, getString(R.string.about));

		binding.appVersion.setText(AppUtils.getVersionName(getApplicationContext()));

		binding.viewLicenses.setOnClickListener(view ->
			new LicensesDialog.Builder(this)
				.setNotices(R.raw.notices)
				.setIncludeOwnLicense(true)
				.setTitle(R.string.licenses)
				.setCloseText(R.string.close)
				.build()
				.show());
	}

	@Override
	public void onBackPressed() {
		finish();
		super.onBackPressed();
	}

	@Override
	public boolean onOptionsItemSelected(MenuItem item) {
		if (item.getItemId() == android.R.id.home) {
			finish();
			return true;
		}

		return super.onOptionsItemSelected(item);
	}
}

File: app/src/main/kotlin/br/com/vitorsalgado/example/features/about/AboutActivity.kt
package br.com.vitorsalgado.example.features.about

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import br.com.vitorsalgado.example.R
import br.com.vitorsalgado.example.utils.AppUtils
import br.com.vitorsalgado.example.utils.delegates.viewWithId
import br.com.vitorsalgado.example.utils.extensions.setToolbar
import de.psdev.licensesdialog.LicensesDialog

class AboutActivity : AppCompatActivity() {
  companion object {
    fun newIntent(context: Context): Intent {
      return Intent(context, AboutActivity::class.java)
    }
  }

  private val toolbar: Toolbar by viewWithId(R.id.toolbar)
  private val appVersion: TextView by viewWithId(R.id.appVersion)
  private val viewLicenses: TextView by viewWithId(R.id.viewLicenses)

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.about_activity)

    setToolbar(toolbar, R.string.about_title)

    appVersion.text = AppUtils.getVersionName(applicationContext)

    viewLicenses.setOnClickListener {
      LicensesDialog.Builder(this)
        .setNotices(R.raw.notices)
        .setIncludeOwnLicense(true)
        .setTitle(R.string.licenses)
        .setCloseText(R.string.close)
        .build()
        .show()
    }
  }

  override fun onBackPressed() {
    finish()
    super.onBackPressed()
  }

  override fun onOptionsItemSelected(item: MenuItem): Boolean {
    if (item.itemId == android.R.id.home) {
      finish()
      return true
    }

    return super.onOptionsItemSelected(item)
  }
}
